src/client/opamCommands.ml

```diff
@@ -710,6 +710,116 @@ let list ?(force_search=false) cli =
+        forest of the specified packages. Note that non-installed packages are \
+        ignored when this option is used.";
+    `P "When a package appears twice or more in the forest, the second or \
+        later occurrences of the said package will be marked as a duplicate, \
+        and be annotated with the $(i,[*]) symbol. Any sub-trees rooted from \
+        such duplicates will be truncated to avoid redundancy.";
+    `P ("See section $(b,"^filter_docs^") and $(b,"^selection_docs^") for all \
+         the ways to select the packages to be displayed, and section \
+         $(b,"^display_docs^") to customise the output format.");
+    `P "For a flat list of packages which may not be installed, \
+        see $(b,opam list).";
+    `S Manpage.s_arguments;
+    `S build_docs;
+    `S filter_docs;
+    `P "These options only take effect when $(i,PACKAGES) are present.";
+    `S selection_docs;
+    `S display_docs;
+  ] in
+  let mode =
+    let default = OpamTreeCommand.(if why then ReverseDeps else Deps) in
+    mk_vflag default ~cli ~section:build_docs [
+      cli_from cli2_2, OpamTreeCommand.Deps, ["deps"],
+      "Draw a dependency forest, starting from the packages not required by \
+       any other packages (this is the default).";
+      cli_from cli2_2, OpamTreeCommand.ReverseDeps, ["rev-deps"],
+      "Draw a reverse-dependency forest, starting from the packages which \
+       have no dependencies.";
+    ]
+  in
+  let filter =
+    let default = OpamTreeCommand.Roots_from in
+    mk_vflag default ~cli ~section:filter_docs [
+      cli_from cli2_2, OpamTreeCommand.Roots_from, ["roots-from"],
+      "Display only the trees which roots from one of the $(i,PACKAGES) \
+       (this is the default).";
+      cli_from cli2_2, OpamTreeCommand.Leads_to,   ["leads-to"],
+      "Display only the branches which leads to one of the $(i,PACKAGES).";
+    ]
+  in
+  let post =
+    mk_flag ~cli (cli_from cli2_2) ["post"] ~section:selection_docs
+      "Include dependencies tagged as $(i,post)."
+  in
+  let dev =
+    mk_flag ~cli (cli_from cli2_2) ["dev"] ~section:selection_docs
+      "Include development packages in dependencies."
+  in
+  let doc_flag =
+    mk_flag ~cli (cli_from cli2_2) ["doc";"with-doc"] ~section:selection_docs
+      "Include doc-only dependencies."
+  in
+  let test =
+    mk_flag ~cli (cli_from cli2_2) ["t";"test";"with-test"]
+      ~section:selection_docs
+      "Include test-only dependencies."
+  in
+  let tools =
+    mk_flag ~cli (cli_from cli2_2) ["with-tools"] ~section:selection_docs
+      "Include development only dependencies."
+  in
+  let no_cstr =
+    mk_flag ~cli (cli_from cli2_2) ["no-constraint"] ~section:display_docs
+      "Do not display the version constraints e.g. $(i,(>= 1.0.0))."
+  in
+  let no_switch =
+    mk_flag ~cli (cli_from cli2_2) ["no-switch"] ~section:selection_docs
+      "Do not take into account switch consistency"
```

Comment:
Is it intelligible?

Reply:
As far as I'm concerned, no :-)

Reply:
new version
```
      "Ignore active switch and simulate installing packages from an empty \
       switch to draw the forest"
```

Reply:
installing packages *to* an empty switch?